File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from "chat_channel"
  end

  def unsubscribed

  end

  def receive(data)
    ActionCable.server.broadcast(
      "chat_channel",
      {
        user_id: current_user.id,
        message: data
      }
    )
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "channels/consumer";

const chatChannel = consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  },
});

// document.addEventListener("click", (e) => {
//   alert("wow");

//   const message = {
//     body: "Hello Wow",
//   };

//   chatChannel.send({ message: "hello wow" });
// });
